### ISSUE TITLE
Block Grid default stylesheet causes problems on safari mobile

### ISSUE THREAD

**Issue by robertjf** [umbraco/Umbraco-CMS#13799]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.1.0

### Bug summary

When viewing a page on Safari in a window less than 640px wide or on an iPhone, the block grid elements in a two-column layout overlap each other - e.g.:

![Screenshot 2023-02-08 at 12 44 10 pm](https://user-images.githubusercontent.com/818101/217407275-9f46cf8d-4933-4fc0-b4dc-c4706b2f9a6a.png)

On a wider screen it looks like this:

![Screenshot 2023-02-08 at 12 45 30 pm](https://user-images.githubusercontent.com/818101/217407449-2e3694fa-9650-4b04-8f28-4c6d6fe1e429.png)

This is not a problem on Chrome based browsers on desktop (MacOS) (But is on iPhone)

It looks like the solution is to change the `.umb-block-grid__area-container` class from this:

```
grid-auto-rows: minmax(50px, min-content);
```

to this:

```
grid-auto-rows: minmax(max-content,min-content);
```

### Specifics

This affects the umbraco-blockgridlayout.css file.

### Steps to reproduce

Add a two-column layout to Block Grid and populate with images - 2 on one side, 1 on the other.

### Expected result / actual result

Images should not be overlaid in the Safari browser on mobile view (<640px)

---
_This item has been added to our backlog [AB#26604](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/26604)_

**Comment by kjac** [comment]
Hi @robertjf,

Thank you for reporting this, and thank you so much for the fix too 💪 

We'll have a look at it and get back to you on the PR.

**Comment by nielslyngsoe** [comment]
Hi @robertjf 

I have followed your Step to reproduce, and thanks for those. Sadly your suggestion makes no difference to me. :-(

What I discovered here is that the images can overlap, because the overflowing parts of the image is not hidden away.
This is cause by the image block's image is set to object-fit: cover; which is why the image is scaled so its covering the space of its block. and exceeds its edges.

This can be fixed by setting `overflow:hidden;`on the image block.

But what you describe seems like another problem, if I'm not mistaken? But it might be related. I hope you will be able to bring back a different way to reproduce, maybe share some of the code of the project you have screenshots from?

In regards to your suggested fix, I will try to describe the current choice of `grid-auto-rows: minmax(50px, min-content);`, just to get some perspective.

This should ensure that the height of the rows are minimum 50px. Mainly to guarantee that Blocks has a proper size so they can be interacted with by the mouse cursor. Mainly for backoffice.

That was the minimum value, for the maximum value we use `min-content` this sets the row height to the smallest available. This helps avoiding a possible large block pushing the rows height to its maximum size. Instead blocks that has a dynamic height will aim for the smallest possible height for all the blocks of that row.

Maybe your Blocks are styled in a way that enables them to overflow the 'container', and the problem lies within that?

As well I'm not sure your suggestion for `grid-auto-rows: minmax(max-content,min-content);` does exactly what you wanted. In theory that should be equivalent to `grid-auto-rows: max-content;`.

I hope that helps, and I'm looking forward to hear back on this issue.
